esmf_regrid/tests/unit/esmf_regridder/test_Regridder.py

```diff
@@ -66,16 +66,15 @@ def test_Regridder_regrid():
     # Set up the regridder with precomputed weights.
```

Comment:
In order to understand, and hence check that we have matched up the esmf computed weights to the correct location in our Python array, I have created a slightly simpler (3x2) array test, that can be worked through manually and checked on paper!   In the following example.  Having achieved this I am confident that we have the somewhat intricate conversions from esmf weight location to Iris location correct (referenced by 'F' ordering, but unravelled in a 'C' ordering, with elements reference per target-src point ordering)!

Given that this conversion is non-trivial to think through understnad and correct, I was wondering if we could include this simpler example alongside the more complex one already implemented.  Note that it is not aiming to check the entire initialisation method, but more explicitly obtaining the weights and converting them into use with the following code.

I think it would have been ideal to re-order the weights into our "Iris format" idexes at this point, rather than flattening the data in the Fortan ordering during the regridding.  However, I have confirmed that the results are correct offline.  (I'll add a note about this in another comment).  Therefore, if we could add this extra test here, I think the code is good to merge, and further improvements could be made once the correct test results are committed.

def test_Regridder_init_small():
    lon_bounds = np.array([0, 10, 20])
    lat_bounds = np.array([0, 10, 20, 30])
    lon, lat = _get_points(lon_bounds), _get_points(lat_bounds)
    src_grid = GridInfo(lon, lat, lon_bounds, lat_bounds)
    assert (src_grid.shape == (3,2))
    assert (src_grid._index_offset() == 1)

    lon_bounds = np.array([0, 20])
    lat_bounds = np.array([0, 10, 30])
    lon, lat = _get_points(lon_bounds), _get_points(lat_bounds)
    tgt_grid = GridInfo(lon, lat, lon_bounds, lat_bounds)
    assert (tgt_grid.shape == (2, 1,))
    assert (tgt_grid._index_offset() == 1)

    rg = Regridder(src_grid, tgt_grid)

    result = rg.weight_matrix
    
    weights_dict = {}
    weights_dict['row_dst'] = np.array([1, 1, 1, 1, 2, 2, 2, 2], dtype=np.int32) - src_grid._index_offset()
    weights_dict['col_src'] = np.array([1, 2, 4, 5, 2, 3, 5, 6], dtype=np.int32) - tgt_grid._index_offset()
    weights_dict['weights'] = np.array([0.4962897, 0.0037103, 0.4962897, 0.0037103 , 0.25484249, 0.24076863, 0.25484249, 0.24076863])

    expected_weights = scipy.sparse.csr_matrix((weights_dict['weights'], (weights_dict['row_dst'], weights_dict['col_src']))) #, shape=shape)
    assert np.allclose(result.toarray(), expected_weights.toarray())

Reply:
Added with additional documentation.